src/Server/Workspace.php

```diff
@@ -200,4 +214,44 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * @param Options|null $settings
+     */
+    public function didChangeConfiguration(Options $settings = null)
+    {
+        if ($settings === null) {
+            return;
+        }
+
+        $changedOptions = $this->getChangedOptions($settings);
+        $this->options = $settings;
+
+        if (!empty(array_intersect($changedOptions, $this->options->getIndexerOptions()))) {
```

Comment:
Yes right now only the `fileTypes` ist implemented.
Later we will have more that trigger a reindex and this part will prevent reindexing more than once at a time.

`getIndexerOptions()` can move somewhere else or be defined as variable inside`didChangeConfiguration`.